examples_notebooks/community_contrib/couchbase/couchbasedb_demo.py

```diff
@@ -0,0 +1,264 @@
+                "%(table_name)s file not found. Setting %(table_name_lower)s to None.",
+                {"table_name": table_name, "table_name_lower": table_name.lower()}
+            )
+            return None
+
+    data["entities"] = load_table("ENTITY_TABLE", read_indexer_entities, pd.read_parquet(f"{INPUT_DIR}/{TABLE_NAMES['ENTITY_EMBEDDING_TABLE']}.parquet"), COMMUNITY_LEVEL)
+    data["relationships"] = load_table("RELATIONSHIP_TABLE", read_indexer_relationships)
+    data["covariates"] = load_table("COVARIATE_TABLE", read_indexer_covariates)
+    data["reports"] = load_table("COMMUNITY_REPORT_TABLE", read_indexer_reports, pd.read_parquet(f"{INPUT_DIR}/{TABLE_NAMES['ENTITY_TABLE']}.parquet"), COMMUNITY_LEVEL)
+    data["text_units"] = load_table("TEXT_UNIT_TABLE", read_indexer_text_units)
+
+    logger.info("Data loading completed")
+    return data
+
+
+def setup_vector_store() -> CouchbaseVectorStore:
+    """Set up and connect to CouchbaseVectorStore."""
+    logger.info("Setting up CouchbaseVectorStore")
+    try:
+        description_embedding_store = CouchbaseVectorStore(
+            collection_name="entity_description_embeddings",
+            bucket_name=COUCHBASE_BUCKET_NAME,
+            scope_name=COUCHBASE_SCOPE_NAME,
+            index_name=COUCHBASE_VECTOR_INDEX_NAME,
+        )
+        description_embedding_store.connect(
+            connection_string=COUCHBASE_CONNECTION_STRING,
+            username=COUCHBASE_USERNAME,
+            password=COUCHBASE_PASSWORD,
+        )
+        logger.info("CouchbaseVectorStore setup completed")
+    except Exception:
+        logger.exception("Error setting up CouchbaseVectorStore:")
+        raise
+
+    return description_embedding_store
+
+
+def setup_models() -> dict[str, Any]:
+    """Set up LLM and embedding models."""
+    logger.info("Setting up LLM and embedding models")
+    try:
+        llm = ChatOpenAI(
+            api_key=OPENAI_API_KEY,
+            model=LLM_MODEL,
+            api_type=OpenaiApiType.OpenAI,
+            max_retries=20,
+        )
+
+        token_encoder = tiktoken.get_encoding("cl100k_base")
+
+        text_embedder = OpenAIEmbedding(
+            api_key=OPENAI_API_KEY,
+            api_base=None,
+            api_type=OpenaiApiType.OpenAI,
+            model=EMBEDDING_MODEL,
+            deployment_name=EMBEDDING_MODEL,
+            max_retries=20,
+        )
+
+        logger.info("LLM and embedding models setup completed")
+    except Exception as e:
+        logger.exception("Error setting up models: %s", str(e))
+        raise
+
+    return {
+        "llm": llm,
+        "token_encoder": token_encoder,
+        "text_embedder": text_embedder,
+    }
+
+
+def store_embeddings(
+    entities: dict[str, Any] | list[Any], vector_store: CouchbaseVectorStore
+) -> None:
+    """Store entity semantic embeddings in Couchbase."""
+    logger.info("Storing entity embeddings")
+
+    try:
+        if isinstance(entities, dict):
```

Comment:
why are you checking the type a dict , Is not list always.

Reply:
made list always